controllers/provisioners/ekscloudformation/bootstrap.go

```diff
@@ -44,9 +44,26 @@ func (ctx *EksCfInstanceGroupContext) getActiveNodeArns() []string {
 	return arnList
 }
 
-func (ctx *EksCfInstanceGroupContext) updateAuthConfigMap() error {
+func (ctx *EksCfInstanceGroupContext) updateAuthConfigMap(existingAuthMap *corev1.ConfigMap) error {
 	arnList := ctx.getActiveNodeArns()
-	configList := []AwsAuthConfig{}
+
+	existingConfigurations := []AwsAuthConfig{}
+	newConfigurations := []AwsAuthConfig{}
+
+	err := yaml.Unmarshal([]byte(existingAuthMap.Data["mapRoles"]), &existingConfigurations)
+	if err != nil {
+		return err
+	}
+
+	for _, configuration := range existingConfigurations {
+		// if existing config is not part of discovered node ARNs, add it as is
```

Comment:
Any reason not to do an upsert of the existingConfigurations? When an IG is deleted, is it removed from the config map?

Reply:
@tekenstam 
Yes, I think I need to clarify the code a little bit + handle deletions now that we are no longer constructing the configmap from scratch.

Will re-iterate and add a commit soon